src/operations.rs

```diff
@@ -85,11 +111,21 @@ pub fn download_extract_sans_parent(
         }?;
 
     let response_with_pb = pb.wrap_read(response);
-
-    let tar = GzDecoder::new(response_with_pb);
-    let archive = Archive::new(tar);
-    unpack_sans_parent(archive, target_path, levels_to_skip)
-        .with_context(|| format!("Failed to extract downloaded file from url `{}`.", url))?;
+    let context = format!("Failed to extract downloaded file from url `{}`.", url);
+
+    // As of this writing, the Rust `tar` library does not fully implement reading the
+    // POSIX.1-2001 PAX format. This is the default format used by BSD tar, and is thus
+    // the format of the Julia tarballs on BSD systems. There is a PR to add support for
+    // PAX to tar.rs: https://github.com/alexcrichton/tar-rs/pull/298, but as of this
+    // writing, it has not been merged. Thus we'll shell out to command line `tar` on
+    // FreeBSD in the meantime, and unify the approaches if/when support is available.
+    if cfg!(target_os = "freebsd") {
```

Comment:
Ah, I didn't realize this was a runtime check, I assumed it wouldn't be since (IIUC?) all configuration things are known at compile time. It is indeed a runtime check though.

Reply:
Done!